ingest/src/Config.ts

```diff
@@ -0,0 +1,51 @@
+import { Embedder, PageStore, EmbeddedContentStore } from "chat-core";
+import { DataSource } from "./sources/DataSource";
+import { ChunkOptions } from "./embed/chunkPage";
+import { IngestMetaStore } from "./IngestMetaStore";
+
+/**
+  The configuration for ingest.
+
+  You can provide your own configuration to the ingest tool.
+
+  Every property is a function that constructs an instance (synchronously or
+  asynchronously). This allows you to run logic for construction or build async.
+  It also avoids unnecessary construction and cleanup if that field of the
+  config is overridden by a subsequent config.
+ */
+export type Config = {
+  /**
+    The store that contains the ingest meta document.
+
+    The ingest meta document is used by the `all` command to determine when the
+    last successful run was for that ingestion.
+   */
```

Comment:
i could envision us expanding the usage of the meta store for more than the all command. perhaps rephrase to be more generic to take into account future use cases?